src/App.tsx

```diff
@@ -14,11 +18,83 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  selectGoods = (good: string) => {
+    this.setState(((prevState: State) => {
+      return { selectedGoods: [...prevState.selectedGoods, good] };
+    }));
+  };
+
+  removeGoods = (good: string) => {
+    this.setState((prevState => ({
```

Comment:
let's be consistent with type definitions - for select goods you specify `prevState` type, here - you don't.
Better stick to certain convention - either define types or not, but keep this approach across your code.
```suggestion
    this.setState((prevState: State => ({
```